test/rubocop/cop/minitest/assert_same_test.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class AssertSameTest < Minitest::Test
+  def test_registers_offense_when_using_equal
+    assert_offense(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert(expected.equal?(actual))
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_same(expected, actual)`.
+        end
+      end
+    RUBY
+
+    assert_correction(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert_same(expected, actual)
+        end
+      end
+    RUBY
+  end
+
+  def test_registers_offense_when_using_equal_with_method_call
+    assert_offense(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert(obj.expected.equal?(other_obj.actual))
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_same(obj.expected, other_obj.actual)`.
+        end
+      end
+    RUBY
+
+    assert_correction(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert_same(obj.expected, other_obj.actual)
+        end
+      end
+    RUBY
+  end
+
+  def test_registers_offense_when_using_assert_with_the_message
+    assert_offense(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert(expected.equal?(actual), 'message')
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_same(expected, actual, 'message')`.
+        end
+      end
+    RUBY
+
+    assert_correction(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert_same(expected, actual, 'message')
+        end
+      end
+    RUBY
+  end
+
+  def test_registers_offense_when_using_assert_with_heredoc_message
+    assert_offense(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert(expected.equal?(actual), <<~MESSAGE
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_same(expected, actual, <<~MESSAGE)`.
+            message
+          MESSAGE
+          )
+        end
+      end
+    RUBY
+
+    assert_correction(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert_same(expected, actual, <<~MESSAGE
+            message
+          MESSAGE
+          )
+        end
+      end
+    RUBY
+  end
+
+  def test_does_not_register_offense_when_using_assert_same
+    assert_no_offenses(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert_same(expected, actual)
+        end
+      end
+    RUBY
+  end
+
+  def test_does_not_register_offense_when_not_using_equal
+    assert_no_offenses(<<~RUBY)
+      class FooTest < Minitest::Test
+        def test_do_something
+          assert(expected.eq?(actual))
+          assert(expected.eql?(actual))
+          assert(expected.equal(actual))
```

Comment:
The tests for `eq?` and `equal` looks unnecessary because these are not built-in methods.

```suggestion
          assert(expected.eql?(actual))
```